cmd/sriovdp/manager.go

```diff
@@ -131,11 +253,21 @@ func (rm *resourceManager) initServers() error {
 			glog.Infof("no devices in device pool, skipping creating resource server for %s", rc.ResourceName)
 			continue
 		}
+
 		rPool, err := rm.rFactory.GetResourcePool(rc, filteredDevices)
 		if err != nil {
 			glog.Errorf("initServers(): error creating ResourcePool with config %+v: %q", rc, err)
 			return err
 		}
+
+		if rm.useCdi {
+			err = cdi.CreateCDISpec(rm.resourcePrefix, filteredDevices, rPool)
```

Comment:
i think this should be part of  resource server.

each server (if its in cdi mode) will (re)create cdi specs for its resource pool when `triggerUpdate()` is called.

the ResourceManager should not concern itself with cdi specifics

thoughts ?

Reply:
Only ResourceManager has information about filtered devices

Reply:
resource pool contains all the devices for the pool (which are already filtered by selectors)...